.github/workflows/docs.yml

```diff
@@ -1,57 +1,36 @@
+---
 name: Docs
 
-on:
+on:  # yamllint disable-line rule:truthy
   push:
     branches:
       - master
 
-jobs:
-  build:
-    name: Build
-
-    permissions:
-      contents: read
-
-    runs-on: ubuntu-latest
+permissions:
+  contents: read
+  id-token: write
+  pages: write
 
-    steps:
-      - name: Install Nix
-        uses: DeterminateSystems/nix-installer-action@main
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          extra-conf: |
-            extra-experimental-features = nix-command flakes
+concurrency:
+  cancel-in-progress: true
+  group: pages
 
-      - name: Set up cache
-        uses: DeterminateSystems/magic-nix-cache-action@main
-
-      - name: Build docs
-        run: nix -L build github:${{ github.repository }}/${{ github.sha }}#docs
-
-      - name: Prepare docs for upload
-        run: cp -r --dereference --no-preserve=mode,ownership result/ public/
```

Comment:
Unless the situation has changed since the original workflow was written, I think this copy was necessary due to an error with the files being part of the Nix store. (It was either because they were owned by `root`, or because there were symlinks involved, I can't remember exactly)